src/modules/did/did.entity.ts

```diff
@@ -1,19 +1,177 @@
+import {
+  IServiceEndpoint,
+  IDIDDocument,
+  IAuthentication,
+  ILinkedDataProof,
+  IPublicKey,
+} from '@ew-did-registry/did-resolver-interface';
+import { Field, ID, Int, ObjectType } from '@nestjs/graphql';
 import { Column, Entity, PrimaryColumn } from 'typeorm';
+import { JSONObject } from '../../common/json.scalar';
+
+interface ProfileClaim {
+  profile?: any;
+}
+
+interface ENSClaim {
+  claimType?: string;
+  claimTypeVersion?: string;
+}
+
+export interface IClaim extends IServiceEndpoint, ProfileClaim, ENSClaim {}
+
+@ObjectType()
+class AuthenticationSchema implements IAuthentication {
+  @Field(() => Int, { nullable: true })
+  block?: number;
+
+  @Field()
+  publicKey: string;
+
+  @Field()
+  type: string;
+
+  @Field(() => Int)
+  validity: any;
+
+  [key: string]: string | number;
+}
+
+@ObjectType()
+class ProofSchema implements ILinkedDataProof {
+  @Field()
+  created: string;
+
+  @Field()
+  creator: string;
+
+  @Field()
+  signatureValue: string;
+
+  @Field()
+  type: string;
+}
+
+@ObjectType()
+class PublicKeySchema implements IPublicKey {
+  @Field(() => Int, { nullable: true })
+  block: number;
+
+  @Field()
+  controller: string;
+
+  @Field({ nullable: true })
+  ethereumAddress?: string;
+
+  @Field()
+  id: string;
+
+  @Field({ nullable: true })
+  publicKeyBase58?: string;
+
+  @Field({ nullable: true })
+  publicKeyBase64?: string;
+
+  @Field({ nullable: true })
+  publicKeyHex?: string;
+
+  @Field({ nullable: true })
+  publicKeyJwk?: string;
+
+  @Field({ nullable: true })
+  publicKeyMultibase?: string;
+
+  @Field({ nullable: true })
+  publicKeyPem: string;
+
+  @Field({ nullable: true })
+  type: string;
+
+  @Field(() => Int, { nullable: true })
+  validity: any;
+
+  [key: string]: string | number;
+}
+
+@ObjectType()
+class ClaimSchema implements IClaim {
+  @Field(() => Int, { nullable: true })
+  block: number;
+
+  @Field({ nullable: true })
+  claimType?: string;
+
+  @Field({ nullable: true })
+  claimTypeVersion?: string;
+
+  @Field()
+  description: string;
+
+  @Field()
+  hash: string;
+
+  @Field()
+  id: string;
+
+  @Field(() => JSONObject, { nullable: true })
+  profile?: any;
+
+  @Field()
+  serviceEndpoint: string;
+
+  @Field()
+  type: string;
+
+  @Field(() => Int)
+  validity: any;
+
+  [key: string]: string | number;
+}
 
 @Entity()
-export class DIDEntity {
+@ObjectType()
+export class DIDEntity implements IDIDDocument {
```

Comment:
Could we rename this to `DIDDocumentEntity` plz? I think it's what I should have named it initially 😅

Reply:
:) no problem :)